redox/processor_neworder.go

```diff
@@ -242,6 +268,143 @@ func (o *newOrderProcessor) handleCreateAccount(ctx context.Context, order model
+				return nil, fmt.Errorf("unexpected status code %v when creating tagName %s", resp.StatusCode(), tagName)
+			}
+		}
+	}
+
+	resp, err := o.clinics.GetClinicWithResponse(ctx, *match.Clinic.Id)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode() != http.StatusOK {
+		return nil, fmt.Errorf("unexpected status code %vwhen fetching clinic with id %s", resp.StatusCode(), *match.Clinic.Id)
+	}
+
+	existingTags = o.getExistingTags(*resp.JSON200)
+	patientTagIds := clinics.PatientTagIds{}
+	for _, tagName := range tagNames {
+		patientTag, ok := existingTags[tagName]
+		if !ok {
+			// The tag should have been created. If it was deleted in the meantime returning an error will result in a retry
+			return nil, fmt.Errorf("patient tag doesn't exist")
+		}
+		patientTagIds = append(patientTagIds, *patientTag.Id)
+	}
+
+	return &patientTagIds, nil
+}
+
+func (o *newOrderProcessor) replacePatientTags(ctx context.Context, match clinics.EHRMatchResponse, tagIds *clinics.PatientTagIds) error {
+	if tagIds == nil {
+		return nil
+	}
+
+	patient := (*match.Patients)[0]
+	update := clinics.UpdatePatientJSONRequestBody{
+		Email:                          patient.Email,
+		BirthDate:                      patient.BirthDate,
+		FullName:                       patient.FullName,
+		Mrn:                            patient.Mrn,
+		TargetDevices:                  patient.TargetDevices,
+		Tags:                           tagIds,
+	}
+	resp, err := o.clinics.UpdatePatientWithResponse(ctx, *match.Clinic.Id, *patient.Id, update)
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode() != http.StatusOK {
+		return fmt.Errorf("unexpected status code %v replacing tags for patient %s", resp.StatusCode(), *patient.Id)
+	}
+	return nil
+}
+
+func (o *newOrderProcessor) getPatientTagCodes(match clinics.EHRMatchResponse) map[string]struct{} {
+	result := make(map[string]struct{})
+	if match.Settings.Tags.Codes != nil {
+		for _, code := range *match.Settings.Tags.Codes {
+			result[code] = struct{}{}
+		}
+	}
+    return result
+}
+
+func (o *newOrderProcessor) getPatientTagsSeparator(match clinics.EHRMatchResponse) *string {
+	return match.Settings.Tags.Separator
+}
+
+func (o *newOrderProcessor) getTagNamesFromOrder(order models.NewOrder, match clinics.EHRMatchResponse) []string {
+	separator := o.getPatientTagsSeparator(match)
+	codes := o.getPatientTagCodes(match)
+	if len(codes) == 0 {
+		return nil
+	}
+
+	tagNames := make([]string, 0)
+	if order.Order.ClinicalInfo != nil {
+		for _, info := range *order.Order.ClinicalInfo {
+			if info.Code != nil && info.Value != nil {
+				if _, ok := codes[*info.Code]; ok {
+					if separator == nil || *separator == "" {
+						tagNames = append(tagNames, strings.TrimSpace(*info.Value))
+					} else {
+						for _, tag := range strings.Split(*info.Value, *separator) {
+							tagNames = append(tagNames,  strings.TrimSpace(tag))
+						}
+					}
+				}
+			}
+		}
+	}
```

Comment:
```suggestion
    if *order.Order.ClinicalInfo == nil {
        return nil
    }
	for _, info := range *order.Order.ClinicalInfo {
		if info.Code == nil || info.Value == nil {
			continue
		}
		if _, found := codes[*info.Code]; !found {
			continue
		}
		if separator == nil || *separator == "" {
			tagNames = append(tagNames, strings.TrimSpace(*info.Value))
			continue
		}
		for _, tag := range strings.Split(*info.Value, *separator) {
			tagNames = append(tagNames, strings.TrimSpace(tag))
		}
	}
```

Just a suggestion. I find it a bit easier to read, as it doesn't indent as much, and I feel like I don't have to keep as much in my head at once.

Reply:
Thanks for the suggestion, definitely more reasble.